commands/server/start.go

```diff
@@ -72,17 +72,28 @@ func StartCmd(gen AppGenerator, logger log.Logger, home string, args []string) e
 	if err != nil {
 		return errors.Wrap(err, "failed to create a listener")
 	}
+
 	svr.SetLogger(logger.With("module", "abci-server"))
-	err = svr.Start()
-	if err != nil {
-		return errors.Wrap(err, "failed to start a server")
-	}
+
 	done := make(chan bool)
-	cmn.TrapSignal(logger, func() {
+	cleanupCallback := func() {
```

Comment:
Don't you want to delete a file here?

I don't know what the purpose of the done channel is, but at least with TrapSignal, I am highly skeptical that any processing will happen in another goroutine before process termination (eg, done listener will never react)

Reply:
So here's the thing: the cleanup happens in server.Stop()
We had a problem with panics, which was caused by the callBack not being executed.

I have added the recover with that same callback. We don't manipulate any files or connections as it's done by tendermint.